cli/exchange/pattern.go

```diff
@@ -206,7 +206,12 @@ func PatternPublish(org, userPw, jsonFilePath, keyFilePath, pubKeyFilePath, patN
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("the org specified in the input file (%s) must match the org specified on the command line (%s)", patFile.Org, org))
 	}
 	patInput := PatternInput{Label: patFile.Label, Description: patFile.Description, Public: patFile.Public, AgreementProtocols: patFile.AgreementProtocols, UserInput: patFile.UserInput}
-
+	
+	//issue 924: Patterns with no services are not allowed 
+	if patFile.Services == nil || len(patFile.Services) == 0 {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("the input file (%s) must contain services, unable to proceed", patFile.Services))
```

Comment:
please change "the input file (%s)" to "the pattern definition (%s)". Sorry I didnt notice this before.

Reply:
change completed --

****